discovery/azure/azure.go

```diff
@@ -629,3 +664,26 @@ func (client *azureClient) getNetworkInterfaceByID(ctx context.Context, networkI
 
 	return &result, nil
 }
+
+// addToCache will add the network interface information for the specified nicID
+// If the cache is disable we do nothing
+func (d *Discovery) addToCache(nicID string, netInt *network.Interface) {
+	if d.cache == nil {
+		return
+	}
+	random := time.Duration(d.cfg.RefreshInterval)
```

Comment:
Why do you want random instead of 10* refresh interval?

Reply:
Ok I guess I am missing something:

We don't want the item to expire at the same time, because if that happen will hit the api call limit, everytime the cache expire, so to mitigate that I use a random to avoid everything to expire at the same time (within a few sec).